csrc/ir/utils.cpp

```diff
@@ -301,6 +301,16 @@ std::vector<Val*> consumerValsOf(const Val* val) {
   return uniqueEntries<Val>(consumer_vals);
 }
 
+// Return all consumers of val
+std::unordered_set<Val*> allConsumerValsOf(const Val* val) {
```

Comment:
I'd expect this to be implemented using some utilities from StmtSort. @naoyam might know. 

I think the current implementation is suboptimal in that it may `allConsumerValsOf` the same Val* multiple times. But I'm happy to accept the function as is if you can make it a separate host IR utility instead of an ir_util.

Reply:
> I'd expect this to be implemented using some utilities from StmtSort. @naoyam might know.

Ok, makes sense. FWIW, `StmtSort` is called through `allConsumerValsOf` > `consumerValsOf`.

> I think the current implementation is suboptimal in that it may `allConsumerValsOf` the same Val* multiple times.

Correct! Let me fix that.

> But I'm happy to accept the function as is if you can make it a separate host IR utility instead of an ir_util.

Ok, np, I can move it away from ir_utils, I just thought it could help others